registry/server/auth/index.ts

```diff
@@ -0,0 +1,99 @@
+import sessionKnex from 'connect-session-knex';
+import { Express, RequestHandler } from 'express';
+import session from 'express-session';
+import { Logger } from 'ilc-plugins-sdk';
+import passport from 'passport';
+
+import { AuthProviders } from '../authEntities/interfaces';
+import db from '../db';
+import { userContextMiddleware } from '../middleware/userContext';
+import { availableMethodsHandlerFactory } from './handlers/availableMethodsHandler';
+import { localLoginHandlerFactory } from './handlers/localLoginHandler';
+import { logoutHandlerFactory } from './handlers/logoutHandler';
+import { openIdAuthCallbackHandlerFactory } from './handlers/openIdAuthCallbackHandler';
+import { bearerAuthenticationMiddlewareFactory } from './middleware/bearerAuthenticationMiddleware';
+import { initializeOpenIdMiddlewareFactory } from './middleware/initializeOpenIdMiddleware';
+import { rolesMiddlewareFactory } from './middleware/rolesMiddleware';
+import { validateSessionMiddlewareFactory } from './middleware/validateSessionMiddleware';
+import { authService } from './services/AuthService';
+import { OpenIdService } from './services/OpenIdService';
+import { bearerStrategyFactory } from './strategies/bearer';
+import { localStrategyFactory } from './strategies/local';
+
+type SetupAuthOptions = {
+    session?: session.SessionOptions;
+};
+
+export async function useAuth(
+    app: Express,
+    openIdService: OpenIdService,
+    config: SetupAuthOptions,
+    logger: Logger,
+): Promise<RequestHandler[]> {
+    const SessionKnex = sessionKnex(session);
+    const sessionConfig = Object.assign(
+        {
+            resave: false,
+            saveUninitialized: false,
+            cookie: { httpOnly: true, secure: false },
```

Comment:
We already switched to use https everywhere, is there a reason to leave secure: false ?

Reply:
Added on top level for production only.